velox/exec/Window.cpp

```diff
@@ -88,6 +88,27 @@ Window::Window(
 Window::WindowFrame Window::createWindowFrame(
     core::WindowNode::Frame frame,
     const RowTypePtr& inputType) {
+  if (frame.type == core::WindowNode::WindowType::kRows) {
+    auto frameBoundCheck = [&](const core::TypedExprPtr& frame) -> void {
+      if (frame == nullptr) {
+        return;
+      }
+      auto frameChannel = exprToChannel(frame.get(), inputType);
+      if (frameChannel == kConstantChannel) {
+        VELOX_USER_CHECK_EQ(
+            frame->type(),
+            BIGINT(),
+            "constant k in frame bounds must be BIGINT type");
```

Comment:
Why are the type requirements are different between constant and non-constant case? Seems strange. Also, should we document these requirements in the WindowNode ?

Reply:
Presto doesn't use constant bounds. They were added so that its not a limitation of Velox WindowNode. 

Might be worth checking with the Spark team about their usage and enhance this accordingly.

Adding documentation to the WindowNode.

Reply:
Added support for int32 constant bounds also.